tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_browser_Browser.java

```diff
@@ -518,6 +518,18 @@ public void changing(LocationEvent event) {
 	for (int i = 0; i < 100; i++) browser.removeLocationListener(listener);
 }
 
+
+
+@Test
+public void test_isLocationForCustomText() {
```

Comment:
Please extend the tests to not only cover the happy path. I consider at least the following scenarios relevant (maybe there are even more):
- check that location after initialization is not the location for custom text
- check that location after moving to some "actual" URL is not considered a location for custom text
- check that when calling `setText()` first and then navigating to some other page via `setUrl()` results in the location not being considered the location for custom text
- check that when calling `setText()` twice and using the browser Url after the first call of `setText()` after the second call of `setText()` to check whether it is a location for custom text still yields true (this is for the case that some asynchronous navigation is performed while some consumer still wants to check the previous URL)
- validate what happens if calling that method after the browser has been disposed

Reply:
I see, So probably the first proposed test case does not make any sense. No need to add a different kind of test for that. This should only be about testing the contract of the new `isLocationForCustomText()` method.

Reply:
Added the tests except for the 4th point. I don't get it. Can you explain a bit?